### PR TITLE
Fix failing tests on ArchLinux and potentially MacOS

### LINKED ISSUE #7567: Hexdigit upper/lower case mismatch on some systems
According to @aarlt some tests fail on macOS like this:

```
yulInterpreterTests/access_large_memory_offsets.yul: FAIL
  Contract:
    {
        mstore(0, 7)
        sstore(0, mload(0))
        mstore(sub(0, 1), sub(0, 1))
        sstore(1, mload(sub(0, 1)))
    }

  Expected result:
    Trace:
    Memory dump:
         0: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff07
      FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE0: 00000000000000000000000000000000000000000000000000000000000000ff
    Storage dump:
      0000000000000000000000000000000000000000000000000000000000000000: 0000000000000000000000000000000000000000000000000000000000000007
      0000000000000000000000000000000000000000000000000000000000000001: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
  Obtained result:
    Trace:
    Memory dump:
         0: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff07
      ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe0: 00000000000000000000000000000000000000000000000000000000000000ff
    Storage dump:
      0000000000000000000000000000000000000000000000000000000000000000: 0000000000000000000000000000000000000000000000000000000000000007
      0000000000000000000000000000000000000000000000000000000000000001: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```

On the latest arch linux I actually have the same problem, so I suspect it's a problem with a new version of a dependency.

### PR DESCRIPTION
**Pull request by Marenz**
fixes #7567

There is one more test that fails at Assembler.cpp:130 but I found no obvious way to fix that one yet as the output is from within jsoncpp.
The problem is again that the output on my system is lower case but the test expects upper case.

### CODE REVIEW

**Review by chriseth — DISMISSED** [review]

File: libevmasm/Instruction.cpp [hunk 1]
```diff
@@ -356,13 +356,13 @@ string dev::eth::disassemble(bytes const& _mem)
 	stringstream ret;
 	eachInstruction(_mem, [&](Instruction _instr, u256 const& _data) {
 		if (!isValidInstruction(_instr))
-			ret << "0x" << hex << int(_instr) << " ";
+			ret << "0x" << uppercase << hex << int(_instr) << " ";
```
Comment (axic) [inline libevmasm/Instruction.cpp]: I know the `std` namespace is imported here, but I find the explicit prefix helpful for `uppercase` and `hex`, etc.

**Comment by aarlt** [comment]
I did a very similar fix, but it seem not to work - I just defaulted to `nouppercase`. See https://github.com/ethereum/solidity/pull/7463/files

I also changed `libdevcore/Common.h`, `libdevcore/CommonData.h`, `libdevcore/FixedHash.h`, `libyul/AsmPrinter.cpp` and `test/tools/yulInterpreter/Interpreter.cpp` - but probably those changes are not really needed.

**Comment by Marenz** [comment]
I used uppercase because that's what the tests originally expected.. I can't see the test results of your linked PR. How did it fail?

**Comment by aarlt** [comment]
@Marenz see https://github.com/ethereum/solidity/pull/7463#issuecomment-533315230 and https://circleci.com/gh/ethereum/solidity/180728?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by Marenz** [comment]
The one involving json is because we can't easily influence how jsoncpp outputs things, the other ones look like they should be solvable the same way we did the others, using `std::uppercase`

File: libevmasm/Instruction.cpp (hunk 1 above)
Comment (Marenz) [reply libevmasm/Instruction.cpp]: Added the `std` for you. I kinda felt the same, but I usually try to keep the original code as much as possible when doing changes ;)

**Comment by Marenz** [comment]
updated: using std:: everywhere now.

**Review by chriseth — APPROVED** [review]

**Pull request merged**